File: Codechef/Marbles/src/marbles/Marbles.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package marbles;

import java.util.Scanner;

/**
 *
 * @author Chotu
 */
class Marbles {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt(); sc.nextLine();
        while (t -- > 0) {
            long n = sc.nextInt(); long r = sc.nextInt(); sc.nextLine();
            long x = n - 1;
            long num = 1;
            long y = ((r-1)>(n-r)) ? (r-1) : (n-r);
            long indx = 0;
            for (long i = x; i > y; i --) {
                indx++;
                num = num*i;
                num = num / indx;
            }
            System.out.println(num);
        }
    }
    
} 

File: practice/marbles.py
t = int(raw_input());
while(t > 0):
	n =int(raw_input());
	r =int(raw_input());
	x = n-1
	num = 1
	y = 0
	if((r-1)>(n-r)):
		y = r-1
	else:
		y = n-1
	indx = 0
	# i = x
	# while(i > y):
	for i in xrange(x,0,y):
		pass
		indx = indx + 1
		num = num*i
		num = num / indx
		# i += -1
	print(num)
	t += -1
